Razmik_Kirakosyan/web/html_css/pagination/page.js

```diff
@@ -0,0 +1,87 @@
+class Pagination {
+	new_array=[];
+	current_page = 0
+	pagesize = 0;
+	div = document.getElementById("index")
+	div1 = document.getElementById("index1")
+	next = document.getElementById("next")
+	previous = document.getElementById("prev")
+	page (array=[],pagesize=0) {
+		this.pagesize=pagesize;
+		console.log(array.length);
+		for(let i = 0; i < Math.ceil(array.length / this.pagesize); i++){
+				this.new_array[i] = array.slice(i * this.pagesize, i * this.pagesize + this.pagesize)
+			
+		}
+
+		for (let i = 1; i <= this.pagesize; i++) {
+			let p = document.createElement("p");
+			p.id = i;
+			this.div1.appendChild(p);
+		}
+
+		for (let i = 1; i <= Math.ceil(array.length / this.pagesize); i++) {
+			let x = document.createElement("button")
+			x.id = i;
+			x.innerHTML = i;
+			x.onclick = (event) => {
+				this.current_page = event.target.id-1;
+				this.change_context()
+			}
+			this.div.insertBefore(x,this.next);
+			x.style.width = "30px";
+			x.style.height = "20px";
+			x.style.marginRight = "10px";	
+			}
+
+		this.change_context()
+	}
+
+	change_context() {
+		for (let i = 1; i <= this.pagesize; i++) {
```

Comment:
We can start iteration from 0 and access this.new_array item with [i] and document.getElementById(i + 1).
And please rename new_array :D